src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,49 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private final Map<Class<?>, Object> instances = new HashMap<>();
+
+    private Injector() {
+    }
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        if (!interfaceClazz.isAnnotationPresent(Component.class)) {
+            throw new RuntimeException("Injection failed, missing @Component annotation on the class "
+                    + interfaceClazz.getName());
+        }
+        if (instances.containsKey(interfaceClazz)) {
+            return instances.get(interfaceClazz);
+        }
+        Object instance = createInstance(interfaceClazz);
+        instances.put(interfaceClazz, instance);
+        return instance;
+    }
+
+    private Object createInstance(Class<?> clazz) {
+        try {
+            Object instance = clazz.getDeclaredConstructor().newInstance();
```

Comment:
This line assumes that the class has a no-argument constructor. If the class does not have such a constructor, this will throw an exception. Consider adding a check or handling for classes without a no-argument constructor.